docs/configuration.rst

```diff
@@ -405,6 +405,61 @@ Use ``style_start`` and ``style_end`` like this::
       }
    ]
 
+.. _needs_filter_data:
+
+needs_filter_data
+~~~~~~~~~~~~~~~~~
+
+This option allows to custom define extra filter.
+
+Configuration example::
+
+   def custom_defined_func():
+       return "my_tag"
+
+   needs_filter_data = {
+       "current_variant": "project_x",
+       "sphinx_tag": custom_defined_func(),
+   }
+
+
+The defined ``needs_filter_data`` must be a dictionary. Its values can be a string variable or a custom defined 
+function. The function get execued during config loading and must return a string. 
+
+Hence, this is **not allowed**::
+
+   needs_filter_data = {
+       "sphinx_tag": custom_defined_func,  # () missing to execute the function
+   }
+
+
+The value of ``needs_filter_data`` will be used as filter string and can be very poweful togehter with 
```

Comment:
No, the values of needs_filter_data will be available as data inside filter_strings.

They do not filter anything or even contain filters. Its just data for comparisons.